src/sentry/integrations/slack/message_builder/issues.py

```diff
@@ -109,9 +109,9 @@ def build_assigned_text(identity: RpcIdentity, assignee: str) -> str | None:
     except ObjectDoesNotExist:
         return None
 
-    if actor.is_team:
+    if actor.is_team and isinstance(assigned_actor, Team):
         assignee_text = f"#{assigned_actor.slug}"
-    elif actor.is_user:
+    elif actor.is_user and isinstance(assigned_actor, RpcUser):
```

Comment:
could probably just use the isinstance checks here since they should do the same thing